### DISCUSSION TITLE
This is fascinating - I have so many questions...

### DISCUSSION THREAD

**Discussion by ichibanchef** [jbogard/ContosoUniversityDotNetCore-Pages#65] [Q&A]
...but I won't ask all of them

In your real world implementations of Vertical Slice Architecture, is this typical; having all the logic for a feature wrapped up in a single class, or do you typically break it out more? Would breaking it out end up defeating the purpose?

For more complex views where users can perform multiple actions, are those broken up into separate components, with all the appropriate logic included, like the razor pages?

**Comment by jbogard** [comment]
Typically no, I'd break those out if they got big. Single file is a preference/convenience thing.

For complex views, that completely depends. Usually I'll apply refactoring to the handlers to break things out into smaller classes/concerns. But it really really depends on the screen itself.
